File: src/main/java/com/github/bpazy/eql/directory/EqlMethodDirectory.java
package com.github.bpazy.eql.directory;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

/**
 * @author ziyuan
 * created on 2017/12/21
 */
public enum EqlMethodDirectory {
    select,
    selectFirst,

    insert,

    delete,

    update;

    public static String toPatternString() {
        List<String> directories = Stream.of(values()).map(EqlMethodDirectory::name).collect(Collectors.toList());
        return String.join("|", directories);
    }
}

File: src/main/java/com/github/bpazy/eql/directory/EqlMethodDirectory.kt
package com.github.bpazy.eql.directory

/**
 * @author ziyuan
 * created on 2017/12/21
 */
enum class EqlMethodDirectory {
    select,
    selectFirst,

    insert,

    delete,

    update;


    companion object {

        fun toPatternString(): String {
            val directories = values().map { v -> v.name }
            return directories.joinToString("|")
        }
    }
}
